gazelle/cue/config.go

```diff
@@ -0,0 +1,150 @@
+package cuelang
+
+import (
+	"flag"
+	"fmt"
+	"go/build"
+	"log"
+	"path"
+	"strings"
+
+	"github.com/bazelbuild/bazel-gazelle/config"
+	"github.com/bazelbuild/bazel-gazelle/rule"
+)
+
+type cueConfig struct {
+	// prefix is a prefix of an import path, used to generate importpath
+	// attributes. Set with -go_prefix or # gazelle:prefix.
+	prefix    string
+	prefixRel string
+	// enableTnargRulesCue indicates whether the tnarg_rules_cue should be enabled.
+	enableTnargRulesCue bool
+
+	// cueTestGoldenSuffix is the suffix used for golden test files in CUE tests.
+	// This allows customizing the suffix for generated test files.
+	// set with #gazelle:cue_test_golden_suffix <FILE_SUFFIX>
+	// example: #gazelle:cue_test_golden_suffix -gen.yml
+	cueTestGoldenSuffix string
+
+	// cueTestGoldenFilename is the filename used for golden test files in CUE tests.
+	// This allows specifying a custom filename for the golden test file.
+	// set with #gazelle:cue_test_golden_filename <FILENAME>
+	// example: #gazelle:cue_test_golden_filename main.gen.json
+	cueTestGoldenFilename string
+
+	// cueGenExportedInstance controls whether to generate cue_exported_instance rules
+	// for each cue_instance. When true, a corresponding cue_exported_instance rule will be created.
+	// #gazelle:cue_exported_instance
+	cueGenExportedInstance bool
+
+	// cueOutputFormat specifies the output format for CUE exports.
+	// Valid values are "json", "yaml", and "text".
+	// Default is "json" if not specified.
+	cueOutputFormat string
+}
+
+// KnownDirectives returns a list of directive keys that this
+// Configurer can interpret. Gazelle prints errors for directives that
+// are not recognized by any Configurer.
+func (s *cueLang) KnownDirectives() []string {
+	return []string{"prefix", "cue_enable_tnarg_rules_cue", "cue_test_golden_suffix", "cue_test_golden_filename", "cue_output_format", "cue_gen_exported_instance"}
```

Comment:
`cue_enable_tnarg_rules_cue`  should be removed since we have been migrate to `rules_cue`